### PR TITLE
[4.0] Remove com_mailto

### PR DESCRIPTION
**Pull request by brianteeman**
with #24025 we removed the mail a link to a friend functionality. But I didnt remove the com_mailto component. Afaict this component was only used for the mail to friend functionality so I have now removed all traces of the component (I hope)

### CODE REVIEW

**Comment by C-Lodder** [comment]
What about those migrating from 3.x to 4.0? Doesn't the `com_mailto` directory need removing too?

**Comment by brianteeman** [comment]
I "think" the pr does that. For the migrating users then it will be handled by script.php at update time

File: administrator/components/com_content/Service/HTML/Icon.php [hunk 1]
```diff
@@ -101,23 +101,7 @@ public function create($category, $params, $attribs = array(), $legacy = false)
 	 */
 	public function email($article, $params, $attribs = array(), $legacy = false)
 	{
-		$uri      = Uri::getInstance();
-		$base     = $uri->toString(array('scheme', 'host', 'port'));
-		$template = $this->application->getTemplate();
-		$link     = $base . Route::_(\ContentHelperRoute::getArticleRoute($article->slug, $article->catid, $article->language), false);
-		$url      = 'index.php?option=com_mailto&tmpl=component&template=' . $template . '&link=' . MailtoHelper::addLink($link);
```
Comment (Quy) [inline administrator/components/com_content/Service/HTML/Icon.php]: Remove `use Joomla\Component\Mailto\Site\Helper\MailtoHelper;`

Comment (brianteeman) [reply administrator/components/com_content/Service/HTML/Icon.php]: fixed thanks

File: administrator/components/com_content/Service/HTML/Icon.php [hunk 2]
```diff
@@ -101,23 +100,7 @@ public function create($category, $params, $attribs = array(), $legacy = false)
...
-
-		$height = Factory::getApplication()->get('captcha', '0') === '0' ? 450 : 550;
-		$status = 'width=400,height=' . $height . ',menubar=yes,resizable=yes';
-
-		$text = LayoutHelper::render('joomla.content.icons.email', array('params' => $params, 'legacy' => $legacy));
-
-		$attribs['title']   = Text::_('JGLOBAL_EMAIL_TITLE');
```
Comment (Quy) [inline administrator/components/com_content/Service/HTML/Icon.php]: Remove language string.

**Comment by zero-24** [comment]
> will be handled by script.php at update time

but the required changes to that file are missing in the script.php right now?

 I have also added the `Documentation required` label as this should be included in the docs too :)

**Comment by brianteeman** [comment]
> but the required changes to that file are missing in the script.php right now?

I meant that the script.php will be updated near to release time in one go. If you noticed you would see that there have not been any commits to that file during the development of J4 with individual PR

> I have also added the Documentation required label as this should be included in the docs too :)

What documentation

**Comment by zero-24** [comment]
https://docs.joomla.org/Potential_backward_compatibility_issues_in_Joomla_4

**Comment by zero-24** [comment]
Maybe an dedicated page why or better that we removed it makes sense too.

**Comment by brianteeman** [comment]
its not a bc issue

better to spend your time making security headers actually work

**Comment by zero-24** [comment]
> better to spend your time making security headers actually work

I'm sorry is there a reason to personally attact me? Can you please tell me what is the problem at the security headers? As there are already bigger changes planed when the current PR gets merged, I can check on the issues too.

**Comment by brianteeman** [comment]
it doesnt work at all.

**Comment by zero-24** [comment]
> it doesnt work at all.

hmm please share your configuration and the steps to reproduce with me. I have just tried it on php 7.3 with the latest build and it worked correctly, but I'm happy to debug and improve what ever is needed. Thanks :)

**Comment by brianteeman** [comment]
see #25592 had to guess what to do as I couldnt find any documentation

**Comment by uglyeoin** [comment]
@brianteeman please can you provide testing instructions and i will test this PR so it can be accepted.  I assume I can simply apply this patch then search in "manage extensions" for "mail to" and if it does not exist I can pass this PR?

**Comment by uglyeoin** [comment]
Do we need to install sample data?  I get this error: 

> The file marked for modification does not exist: installation/sql/mysql/sample_testing.sql

When trying to install the "testing" sample data I also get another error.  I know that's unrelated but can you confirm it makes this test impossible.  Or is the `sample_testing.sql` part of this pull request unecessary?

**Comment by brianteeman** [comment]
conflicts resolved.

@uglyeoin that will resolve your patchtester error. As for testing instructions you got it right

**Comment by uglyeoin** [comment]
God almighty.  Who do I have to butter up to get that tooltip off the issue number in Patchtester when I try to type it in?  It's destroying my brain.

**Comment by uglyeoin** [comment]
I cannot apply the patch.  

> The patch could not be applied because it conflicts with a previously applied patch: installation/sql/mysql/joomla.sql

I have checked and this is the only patch applied, I have reverted all others.

**Comment by Quy** [comment]
> God almighty. Who do I have to butter up to get that tooltip off the issue number in Patchtester when I try to type it in? It's destroying my brain.

Are you referring to the alert? Try this: https://github.com/joomla/joomla-cms/issues/25850#issuecomment-525044000

**Comment by brianteeman** [comment]
@uglyeoin report that to the patchtester repo. I cannot confirm with the old version of patchtester I use.

**Comment by uglyeoin** [comment]
@quy that isn't the problem, this is the problem.  https://github.com/joomla-extensions/patchtester/issues/243#issuecomment-546618372

I suggested the solution.  I'd do a pull request but I'm really not sure how to do it when it comes to compiled CSS.  If someone can point me at instructions I'll give it a go.

**Comment by brianteeman** [comment]
conflicts resolved

File: administrator/components/com_content/Service/HTML/Icon.php [hunk 3]
```diff
@@ -101,23 +100,7 @@ public function create($category, $params, $attribs = array(), $legacy = false)
 	 */
 	public function email($article, $params, $attribs = array(), $legacy = false)
 	{
-		$uri      = Uri::getInstance();
```
Comment (Quy) [inline administrator/components/com_content/Service/HTML/Icon.php]: Should this be deprecated in v3.10 and remove in v4.0 like you did in #25502?

Comment (brianteeman) [reply administrator/components/com_content/Service/HTML/Icon.php]: I am not sure - ask @wilsonge

Comment (Quy) [reply administrator/components/com_content/Service/HTML/Icon.php]: You had a similar conversation with @wilsonge that resulted in #27067.

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on 3f180e956d7b9476fe00dd8fad7122d028d36165<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25516">issues.joomla.org/tracker/joomla-cms/25516</a>.</sub>

File: administrator/components/com_content/Service/HTML/Icon.php (hunk 3 above)
Comment (wilsonge) [reply administrator/components/com_content/Service/HTML/Icon.php]: Deprecate + remove

**Comment by brianteeman** [comment]
Is there any point in me resolving conflicts again or am I wasting my time

File: administrator/components/com_content/Service/HTML/Icon.php (hunk 3 above)
Comment (Quy) [reply administrator/components/com_content/Service/HTML/Icon.php]: @brianteeman I assume it is go since @wilsonge commented to deprecate & remove.

**Comment by brianteeman** [comment]
conflicts resolved and deprecation pr created

**Comment by zero-24** [comment]
Deprecation PR merged :+1:

**Comment by HLeithner** [comment]
Thanks for removing this awful feature.

**Comment by brianteeman** [comment]
Finally!!!!

**Pull request merged**